bins/fetch/src/main.rs

```diff
@@ -95,28 +96,32 @@ async fn main() -> Result<(), Box<dyn Error>> {
     }
 
     // Write compressed bytecodes to disk.
-    let file_bytecodes = File::create("data/bytecodes.bincode.gz")
+    let writer_bytecodes = File::create("data/bytecodes.bincode.gz")
+        .map(|f| GzEncoder::new(f, Compression::default()))
         .map_err(|err| format!("Failed to create compressed bytecodes file: {err}"))?;
-    let serialized_bytecodes = bincode::serialize(&bytecodes)
-        .map_err(|err| format!("Failed to serialize bytecodes to bincode: {err}"))?;
-    GzEncoder::new(file_bytecodes, Compression::default())
-        .write_all(&serialized_bytecodes)
+    bincode::serialize_into(writer_bytecodes, &bytecodes)
         .map_err(|err| format!("Failed to write bytecodes to file: {err}"))?;
 
     // Write pre-state to disk.
     let file_state = File::create(format!("{block_dir}/pre_state.json"))
         .map_err(|err| format!("Failed to create pre-state file: {err}"))?;
-    let json_state = serde_json::to_value(&state)
-        .map_err(|err| format!("Failed to serialize pre-state to JSON: {err}"))?;
-    serde_json::to_writer(file_state, &json_state)
+    serde_json::to_writer(file_state, &state)
         .map_err(|err| format!("Failed to write pre-state to file: {err}"))?;
 
-    // Write block hashes to disk.
-    let block_hashes: BTreeMap<u64, B256> = storage.get_cache_block_hashes().into_iter().collect();
+    // TODO: Deduplicate logic with [for_each_block_from_disk] when there is more usage
+    let mut block_hashes = match File::open("data/block_hashes.bincode") {
+        Ok(compressed_file) => bincode::deserialize_from(compressed_file)
+            .map_err(|err| format!("Failed to deserialize bytecodes from file: {err}"))?,
+        Err(_) => BTreeMap::<u64, B256>::new(),
```

Comment:
I like this being shorter but having the type in an `Err` arm is not as natural as declaring the type in `let`. I'd prefer to follow `bytecodes` and perhaps use `Default::default()` in this arm instead.